### PR TITLE
docs(nuget): Document lockfile exception

### PR DESCRIPTION
**Pull request by horihel**
## Changes:

as dicussed in #13589 this exception to nuget behaviour should be documented

although this point can be removed if #13591 gets considered.

## Context:

This is to avoid confusion and potentially safe others the time I invested while trying to figure out why this didn't work.

## Documentation (please check one with an [x])

- [X] I have updated the documentation, or

## How I've tested my work (please tick one)

I have verified these changes via:

doc update only

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/nuget.md [hunk 1]
```diff
@@ -49,6 +49,8 @@ In this example we defined 3 NuGet feeds.
 The package resolving process uses the `merge` strategy to handle the 3 feeds.
 All feeds are checked for dependency updates, and duplicate updates are merged/joined together into a single dependency update.
 
+If your project uses lockfiles (a `package.lock.json` exists), configuration settings are ignored and _only_ `NuGet.config` and `https://api.nuget.org/v3/index.json` are considered for lookups.
```
Comment (rarkins) [inline docs/usage/nuget.md:52]: "configuration settings" is too broad and likely to confuse

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/nuget.md (hunk 1 above)
Comment (horihel) [reply docs/usage/nuget.md:52]: you're right - would "above settings" be clearer?

File: docs/usage/nuget.md [hunk 2]
```diff
@@ -49,6 +49,8 @@ In this example we defined 3 NuGet feeds.
 The package resolving process uses the `merge` strategy to handle the 3 feeds.
 All feeds are checked for dependency updates, and duplicate updates are merged/joined together into a single dependency update.
 
+If your project uses lockfiles (a `package.lock.json` exists), alternate feed settings are ignored and _only_ `NuGet.config` and `https://api.nuget.org/v3/index.json` are considered for lookups.
```
Comment (rarkins) [inline docs/usage/nuget.md:52]:
```suggestion
If your project uses lockfiles (a `package.lock.json` exists), alternate feed settings must be defined in a `NuGet.config` as `registryUrls` are not passed through to the NuGet commands used.
```

Is this accurate?

Comment (horihel) [reply docs/usage/nuget.md:52]: It is.

i'd want to add an "only" to make clear that central configuration will not work with lockfiles.

**Review by HonkingGoose — APPROVED** [review]

**Pull request merged**